src/dialects/yugabyte/index.js

```diff
@@ -0,0 +1,154 @@
+          values = Utils.defaults(values, options.where);
+        }
+
+        options.exception = true;
+        options.returning = true;
+
+        try {
+          const created = await this.create(values, options);
+          if (created.get(this.primaryKeyAttribute, { raw: true }) === null) {
+            // If the query returned an empty result for the primary key, we know that this was actually a unique constraint violation
+            throw new sequelizeErrors.UniqueConstraintError();
+          }
+
+          return [created, true];
+        } catch (error) {
+          if (!(error instanceof sequelizeErrors.UniqueConstraintError)) {
+            throw error;
+          }
+
+          const flattenedWhere = Utils.flattenObjectDeep(options.where);
+          const flattenedWhereKeys = Object.keys(flattenedWhere).map(name => _.last(name.split('.')));
+          const whereFields = flattenedWhereKeys.map(name => _.get(this.rawAttributes, `${name}.field`, name));
+          const defaultFields = options.defaults && Object.keys(options.defaults)
+            .filter(name => this.rawAttributes[name])
+            .map(name => this.rawAttributes[name].field || name);
+
+          // YugabyteDB related changes are done at this point for errFieldsKeys variable
+          let errFieldKeys = [];
+          if (error.fields !== undefined) {
+            errFieldKeys = Object.keys(error.fields);
+          }
+
+          const errFieldsWhereIntersects = Utils.intersects(errFieldKeys, whereFields);
+          if (defaultFields && !errFieldsWhereIntersects && Utils.intersects(errFieldKeys, defaultFields)) {
+            throw error;
+          }
+
+          if (errFieldsWhereIntersects) {
+            _.each(error.fields, (value, key) => {
+              const name = this.fieldRawAttributesMap[key].fieldName;
+              if (value.toString() !== options.where[name].toString()) {
+                throw new Error(`${this.name}#findOrCreate: value used for ${name} was not equal for both the find and the create calls, '${options.where[name]}' vs '${value}'`);
+              }
+            });
+          }
+
+          // Someone must have created a matching instance inside the same transaction since we last did a find. Let's find it!
+          const otherCreated = await this.findOne(Utils.defaults({
+            transaction: internalTransaction ? null : transaction,
+          }, options));
+
+          // Sanity check, ideally we caught this at the defaultFeilds/err.fields check
+          // But if we didn't and instance is null, we will throw
+          if (otherCreated === null) {
+            throw error;
+          }
+
+          return [otherCreated, false];
+        }
+      } finally {
+        if (internalTransaction && transaction) {
+          await transaction.commit();
+        }
+      }
+    };
+  }
+
+}
+
+// YugabyteDialect.prototype.defaultVersion = '2.11.1.0-b305 '; // minimum supported version
```

Comment:
Are we planning to uncomment this later?

Reply:
No, I will remove this it is not required.

Reply:
Does postgres use this?